examples/simon.py

```diff
@@ -0,0 +1,339 @@
+        for j in list(str(format(secret_function[0][o], "0"+str(number_qubits)+"b"))):
+            if (int(j) == 0):
+                yield cirq.X.on(first_qubits[counter])
+            counter = counter+1
+        yield apply_n_qubit_tof(ancilla, first_qubits+[the_activator])
+        counter = 0
+        for j in list(str(format(secret_function[0][o], "0"+str(number_qubits)+"b"))):
+            if (int(j) == 0):
+                yield cirq.X.on(first_qubits[counter])
+            counter = counter+1
+
+def apply_n_qubit_tof(ancilla, args):
+
+    if (len(args) == 3):
+        yield cirq.CCX.on(args[0], args[1], args[2])
+
+    else:
+
+        yield cirq.CCX.on(args[0], args[1], ancilla[0])
+        for k in range(2, len(args)-1):
+            yield cirq.CCX(args[k], ancilla[k-2], ancilla[k-1])
+
+        yield cirq.CNOT.on(ancilla[len(args)-3], args[len(args)-1])
+
+        for k in range(len(args)-2, 1, -1):
+            yield cirq.CCX(args[k], ancilla[k-2], ancilla[k-1])
+        yield cirq.CCX.on(args[0], args[1], ancilla[0])
+
+
+def make_simon_circuit(first_qubits, second_qubits, oracle):
+
+    circuit = cirq.Circuit()
+
+    #Apply the first set of Hadamard gates
+
+    for i in range(0, number_qubits):
+        circuit.append(cirq.H.on(first_qubits[i]))
+
+    #Apply the oracle
+
+    circuit.append(oracle)
+
+    #Apply the second set of Hadamard gates
+
+    for i in range(0, number_qubits):
+        circuit.append(cirq.H.on(first_qubits[i]))
+
+    #Perform measurements upon the qubits
+
+    circuit.append(cirq.measure(*second_qubits, key='x'))
+    circuit.append(cirq.measure(*first_qubits, key='y'))
+
+    return circuit
+
+run = main(number_qubits)
+matrix_input = run[0]
+secret_function = run[1]
+string_secret = run[2]
+r = run[3]
+
+def shuffle_op(matrix, point):
+
+
+    for i in range(0, len(matrix)):
+        if (matrix[i] == [0 for l in range(0, len(matrix[0]))]):
+            raise ValueError("System of equations not linearly independent, try again")
+        for j in range(0, len(matrix)):
+            if (matrix[i] == matrix[j] and i != j):
+                raise ValueError("System of equations not linearly independent, try again")
+
+
+    for i in range(1, len(matrix)+1):
+      for c in list(itertools.combinations([y[0:len(matrix)+1] for y in matrix], i)):
+        hol = []
+        for b in c:
+          hol.append(b)
+        calc = [sum(x)%2 for x in zip(*hol)]
+        ha = True
+        for ij in range (0, len(hol)):
+            for ik in range (0, len(hol)):
+                if (hol[ik] == hol[ij] and ik != ij):
+                    ha = False
+        if (ha == True and calc == [0 for p in range(0, len(calc))]):
+          raise ValueError("System of equations not linearly independent, try again")
+
+    flip = False
+    passage = False
+
+    for i in range(0, len(matrix)):
+        for j in range(i+1, len(matrix)):
+            if (matrix[i][i] != 0):
+                x = -1*matrix[j][i]/matrix[i][i]
```

Comment:
I think this block is just `matrix[j] -= matrix[i] * int(matrix[j][i] / matrix[i][i])`, assuming matrix is a numpy array.